Chummer/frmCareer.cs

```diff
@@ -16313,6 +16368,295 @@ private void AddCyberwareSuite(Improvement.ImprovementSource objSource)
 
             IsDirty = true;
         }
+
+        /// <summary>
+        /// Refreshes the Sustained Spells Panel
+        /// </summary>
+        /// <param name="panSustainedSpells"></param>
+        /// <param name="notifyCollectionChangedEventArgs"></param>
+        private void RefreshSustainedSpells(Panel flpSustainedSpells, Panel flpSustainedComplexForms, Panel flpSustainedCritterPowers, NotifyCollectionChangedEventArgs notifyCollectionChangedEventArgs = null)
+        {
+            if (flpSustainedSpells == null && flpSustainedComplexForms == null && flpSustainedCritterPowers == null)
+                return;
+
+            if (CharacterObject.SustainedCollection.Count != 0 && !chkPsycheActive.Visible)
+            {
+                //Sets up the Psyche Active Checkbox
+                chkPsycheActive.Visible = true;
+
+            }
+
+            if(notifyCollectionChangedEventArgs == null)
+            {
+                flpSustainedSpells.SuspendLayout();
+                flpSustainedSpells.Controls.Clear();
+                int intSustainedSpells = -1;
+
+
+                foreach (ISustainable objSustained in CharacterObject.SustainedCollection)
+                {
+                    SustainedSpellControl objSustainedSpellControl = new SustainedSpellControl(objSustained);
+
+                    objSustainedSpellControl.SpellDetailChanged += MakeDirtyWithCharacterUpdate;
+                    objSustainedSpellControl.UnsustainSpell += DeleteSustainedSpell;
+
+                    intSustainedSpells += 1;
+                    objSustainedSpellControl.Top = intSustainedSpells * objSustainedSpellControl.Height;
+
+
+                    if (objSustained is SustainedSpell)
+                        flpSustainedSpells.Controls.Add(objSustainedSpellControl);
+
+                    if (objSustained is SustainedComplexForm)
+                        flpSustainedComplexForms.Controls.Add(objSustainedSpellControl);
+
+                    if (objSustained is SustainedCritterPower)
+                        flpSustainedCritterPowers.Controls.Add(objSustainedSpellControl);
```

Comment:
So this can potentially cause some weirdness if a class somehow inherits from multiple sustained object types. You should probably put `else`s before the latter `if`s.

Reply:
I originally didn't use `switch `s here, because I read they should only be used with >5 cases.

But instead using an `else` I now wrapped them into a switch with a `default`, because they are way easier to read and expand upon imo.
I also moved that one out into it's own `DetermineRefreshingPanle()` to reduce the size of this.